app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb

```diff
@@ -41,12 +42,24 @@ def decoded_attributes
         def decoded_id_token
           @decoded_id_token ||= OpenIDConnect::ResponseObject::IdToken.decode(
             @id_token_jwt,
-            get_cert(@provider_uri)
+            get_cert
           )
+        rescue => e
+          raise IdTokenInvalidFormat, e.to_s
+        end
+
+        def discovered_provider
+          @discovered_provider ||= OpenIDConnect::Discovery::Provider::Config.discover!(@provider_uri)
+        rescue HTTPClient::ConnectTimeoutError => e
+          raise ProviderDiscoveryTimeout, @provider_uri
+        rescue => e
+          raise ProviderDiscoveryFailed, @provider_uri
         end
 
-        def get_cert(provider_uri)
-          OpenIDConnect::Discovery::Provider::Config.discover!(provider_uri).jwks
+        def get_cert
+          discovered_provider.jwks
+        rescue => e
```

Comment:
Any chance we get a useful error here that might tell us what went wrong? If so, we should try to preserve it.